packages/repack/src/server/DevServerProxy.ts

```diff
@@ -282,17 +282,12 @@ export class DevServerProxy extends BaseDevServer {
           });
           reply.code(400).send();
         } else {
-          try {
-            if (this.workers[platform]) {
-              await this.forwardRequest(platform, request, reply);
-            } else {
-              await this.runWorker(platform);
-              await this.forwardRequest(platform, request, reply);
-            }
-          } catch (error) {
-            console.error(error);
-            reply.code(500).send();
-          }
+          Promise.resolve(!this.workers[platform] && this.runWorker(platform))
+            .then(() => this.forwardRequest(platform, request, reply)) // eslint-disable-line
+            .catch((error) => {
+              console.error(error);
+              reply.code(500).send();
+            });
```

Comment:
What's the motivation for this change? It doesn't fix anything and makes the code less readable. I'd suggest to revert it.

Reply:
The motivation for this workaround was a bug I ran into while trying to implement `reply.from()` in the `forwardRequest` method.

> Promise may not be fulfilled with 'undefined' when statusCode is not 204

As I understood the error, within an async handler, you need to ensure that all data is sent prior to the handler resolving. TS shows that the `reply.from`, now implemented within `forwardRequest`, returns void, which means it cannot be awaited. I assumed that meant the internals of the function would call send.  My solution then was to modify the handler to not return a promise and instead complete the request via the send method called within `reply.from`.

Revisiting the issue after your comment, and perhaps there is another solution: https://github.com/fastify/fastify-reply-from/issues/105#issuecomment-707696530.

TBH, I'm not entirely sure why it would work if the `reply.from` returns void, but perhaps this is an issue with the types and I can actually await the `reply.from` call. I'll have to investigate further if this is a blocking issue for you.

Reply:
Ok so it's more nuanced than I thought.
Based on the docs: https://www.fastify.io/docs/latest/Routes/#async-await, more specifically section about returning reply:
```js
fastify.get('/', options, async function (request, reply) {
  setImmediate(() => {
    reply.send({ hello: 'world' })
  })
  return reply
})
```
and the comment you've posted I think that this should works:
```js
async (request, reply) => {
  // ... rest of the code
  reply.from(...)
  return reply;
}
```

Reply:
Would this work:
```js
async forwardRequest(...) {
  // ...
  reply.from(...)
  return reply;
}
// ... 
handler: async (request, reply) => {
  // ...
  return await this.forwardRequest(...)
}
```
??

Reply:
Alright, I just pushed updates containing the requested changes.

According to the docs of fastify, `decorateReply()` which is what **fastify-reply-from** utilizes, [sets the context to the Reply instance](https://www.fastify.io/docs/latest/Decorators/#decoratereplyname-value-dependencies). And looking at the source code of **fastify-reply-from**, the return value of `from()` [is actually `this`](https://github.com/fastify/fastify-reply-from/blob/069ee15af3797c7b75a42d90019711efba25c901/index.js#L158), [not void](https://github.com/fastify/fastify-reply-from/blob/069ee15af3797c7b75a42d90019711efba25c901/index.d.ts#L63). In short, this means I can update the handler to return`this.forwardRequest(...)` as long as it returns `reply.from()`.

I have manually tested, and everything appears to be behaving correctly.

One additional note is that the log output from proxying is a little bit noisier than the previous implementation. The following logs are now included and it doesn't appear there is a simple option within **fastify-reply-from** to disable them:

```
ℹ [21:14:10.836Z][DevServerProxy] fetching from remote server {
  source: 'http://localhost:56008/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=app.staple.local'
} 
ℹ [21:14:10.853Z][DevServerProxy] response received 
```